src/increment_version.rs

```diff
@@ -0,0 +1,182 @@
+  old_version: String,
+
+  /// The increment range.
+  #[arg(long, short)]
+  range: crate::VersionRange,
+
+  /// In case of Rust projects:  which package's version shall be edited?
+  #[arg(long, short = 'R')]
+  rust_package: Option<String>,
+}
+
+impl IncrementVersion {
+  /// Process the input data.
+  ///
+  /// # Errors
+  ///
+  /// See
+  ///
+  /// - [`crate::PatternBuffer::try_into_string`]
+  /// - [`PatternReader::read`]
+  /// - [`PatternWriter::truncate`]
+  /// - [`Version::from_str`]
+  /// - [`sysexits::ExitCode::DataErr`]
+  /// - [`sysexits::ExitCode::Unavailable`]
+  /// - [`sysexits::ExitCode::Usage`]
+  pub fn main(&self) -> Result<()> {
+    self.wrap().main()
+  }
+
+  fn wrap(&self) -> Logic {
+    Logic {
+      cli: self.clone(),
+      new_version: String::new(),
+      old_version: self
+        .old_version
+        .strip_prefix('v')
+        .unwrap_or(&self.old_version)
+        .to_string(),
+    }
+  }
+}
+
+struct Logic {
+  cli: IncrementVersion,
+  new_version: String,
+  old_version: String,
+}
+
+impl Logic {
+  fn determine_new_version(&mut self) -> Result<()> {
+    self.new_version = Version::from_str(&self.old_version)?
+      .increment(self.cli.range)
+      .to_string();
+    self.new_version = self
+      .new_version
+      .strip_prefix('v')
+      .unwrap_or(&self.new_version)
+      .to_string();
+    Ok(())
+  }
+
+  fn main(&mut self) -> Result<()> {
+    self.determine_new_version()?;
+
+    for file in &self.cli.file_to_edit {
+      match file
+        .file_name()
+        .ok_or(ExitCode::Usage)?
+        .to_str()
+        .ok_or(ExitCode::DataErr)?
+      {
+        "Cargo.lock" => self.update_cargo_lock(file)?,
+        "Cargo.toml" => self.update_cargo_toml(file)?,
+        _ => self.update_normal_file(file)?,
+      }
+    }
+
+    Ok(())
+  }
+
+  fn update_cargo_lock(&self, file: &PathBuf) -> Result<()> {
+    if let Some(rust_package) = &self.cli.rust_package {
+      let mut edited = false;
+      let mut lock_file = match cargo_lock::Lockfile::load(file) {
+        Ok(l) => Ok(l),
+        Err(cargo_lock::Error::Io(e)) => Err(e.into()),
+        Err(_) => Err(ExitCode::Unavailable),
+      }?;
+
+      for package in &mut lock_file.packages {
+        if package.name.as_str() == rust_package {
+          match self.cli.range {
+            VersionRange::Major => {
+              package.version.major += 1;
+              package.version.minor = 0;
+              package.version.patch = 0;
+            }
+            VersionRange::Minor => {
+              package.version.minor += 1;
+              package.version.patch = 0;
+            }
+            VersionRange::Patch => package.version.patch += 1,
+          }
+
+          edited = true;
+          break;
+        }
+      }
+
+      if edited {
+        file.truncate(Box::new(lock_file.clone().to_string()))
+      } else {
+        Err(ExitCode::DataErr)
+      }
+    } else {
+      Err(ExitCode::Usage)
+    }
+  }
```

Comment:
*Any* file named `Cargo.lock` will be updated as follows:

- Aeruginous searches for the given package name in the respective `Cargo.lock` and updates exclusively that package's version.
- If no such package is given in the respective `Cargo.toml`, the update will fail.
- If *any* file named `Cargo.lock` is given but no package name is specified, the update will fail.